File: API/Connect_API.R
library(httr) # install.packages("httr")

username <- "username"
password <- "password"

query <- "https://api.connect.ihsmarkit.com/dataplatform/v1/odata/ChemicalCapacityByCompany_CapacityToProduce_API_Data_Simplified?$top=3000"

while (!is.null(query) && query!= "") {
  response <- GET(query, authenticate(username, password, type = "basic"))
  stop_for_status(response)
  warn_for_status(response)
  
  response_content <- content(response, encoding = "UTF-8")
  print(response_content)
  
  query <- toString(response_content[["@odata.nextLink"]])
}

File: API/Connect_API.py
import json
import requests
import pandas as pd
from pandas import json_normalize

username = ""
password = ""

# https://api.connect.ihsmarkit.com/swagger/ui/index#/

query = "https://api.connect.ihsmarkit.com/dataplatform/v1/odata/Oil_Markets_Midstream_And_Downstream_API_Data_Simplified"

while query:
    response = requests.get(query, auth=(username, password))
    response.raise_for_status()

    response_content = json.loads(response.content)
    print(response_content)
    
    query = response_content.get("@odata.nextLink", None)